### PR TITLE
add warning when addTabbedWindow is called on self

### PR DESCRIPTION
**Pull request by codebytere**
Closes https://github.com/electron/electron/issues/12046.

Per [documentation](https://developer.apple.com/documentation/appkit/nswindow/1855947-addtabbedwindow?language=objc), you can't add a tabbed window with the same window; you need to create a new one. When you add a new window, you need to pass in a value for `NSWindowOrderingMode`, which can be only `NSWindowAbove`, `NSWindowBelow`, or `NSWindowOut`. As you can't set a window above or below itself, this will crash it. 

This PR catches attempts to do the above and instead outputs an error explaining that it must be called on a new window.

/cc @bpasero @MarshallOfSound

### CODE REVIEW

File: atom/browser/native_window_mac.mm [hunk 1]
```diff
@@ -1672,8 +1672,12 @@ static bool FromV8(v8::Isolate* isolate, v8::Handle<v8::Value> val,
 }
 
 void NativeWindowMac::AddTabbedWindow(NativeWindow* window) {
-  if ([window_ respondsToSelector:@selector(addTabbedWindow:ordered:)]) {
-    [window_ addTabbedWindow:window->GetNativeWindow() ordered:NSWindowAbove];
+  if (window_.get() == window->GetNativeWindow()) {
+    NSLog(@"Error: AddTabbedWindow cannot be called by a window on itself.");
```
Comment (MarshallOfSound) [inline atom/browser/native_window_mac.mm]: This doesn't go over JS logs, can we do `ThrowError` instead (pass in `mate::arguments` or something)

File: atom/browser/api/atom_api_browser_window.cc [hunk 2]
```diff
@@ -1082,8 +1082,9 @@ void BrowserWindow::ToggleTabBar() {
   window_->ToggleTabBar();
 }
 
-void BrowserWindow::AddTabbedWindow(NativeWindow* window) {
-  window_->AddTabbedWindow(window);
+void BrowserWindow::AddTabbedWindow(NativeWindow* window,
+  mate::Arguments* args) {
+  window_->AddTabbedWindow(window, args);
```
Comment (deepak1556) [inline atom/browser/api/atom_api_browser_window.cc]: We shouldn't be processing/using `native-mate` outside the api layer, can you instead change the signature of `NativeWindow::AddTabbedWindow` to have a return value and throw error in `api::BrowserWindow` ? Thanks!

Comment (codebytere) [reply atom/browser/api/atom_api_browser_window.cc]: on it!

File: atom/browser/native_window_mac.mm [hunk 3]
```diff
@@ -918,6 +918,7 @@ static bool FromV8(v8::Isolate* isolate, v8::Handle<v8::Value> val,
   }
 
   if (transparent()) {
+    NSLog(@"Setting transparent");
```
Comment (deepak1556) [inline atom/browser/native_window_mac.mm]: I believe this is not related to the PR.

File: atom/browser/api/atom_api_browser_window.cc [hunk 4]
```diff
@@ -1082,8 +1082,11 @@ void BrowserWindow::ToggleTabBar() {
   window_->ToggleTabBar();
 }
 
-void BrowserWindow::AddTabbedWindow(NativeWindow* window) {
-  window_->AddTabbedWindow(window);
+void BrowserWindow::AddTabbedWindow(NativeWindow* window,
+  mate::Arguments* args) {
```
Comment (deepak1556) [inline atom/browser/api/atom_api_browser_window.cc]: Can you run `clang-format` to fix style issue.

File: atom/browser/native_window_mac.mm (hunk 3 above)
Comment (MarshallOfSound) [reply atom/browser/native_window_mac.mm]: Woops, had that in my staged env 🏃
